---
 automake.in |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: automake.in
===================================================================
--- automake.in.orig	2013-06-03 16:18:00.000000000 +0200
+++ automake.in	2013-06-12 15:04:11.772901297 +0200
@@ -7489,12 +7489,14 @@ sub required_file_check_or_copy ($$$)
   my $fullfile = "$dir/$file";
   my $found_it = 0;
   my $dangling_sym = 0;
+  my $thisdir = dirname ($fullfile);
+  my $thisfile = basename ($fullfile);
 
   if (-l $fullfile && ! -f $fullfile)
     {
       $dangling_sym = 1;
     }
-  elsif (dir_has_case_matching_file ($dir, $file))
+  elsif (dir_has_case_matching_file ($thisdir, $thisfile))
     {
       $found_it = 1;
     }
