--- a/Makefile.org	2007-11-30 14:01:09.000000000 +0100
+++ b/Makefile	2014-01-01 20:38:07.058838305 +0100
@@ -30,6 +30,7 @@
 	 CFLAGS = $(RPM_OPT_FLAGS) $(COPTS) $(DEBUG) $(INC) -D_GNU_SOURCE -Wall -pipe
 	  CLOOP = -funroll-loops
 	     CC = gcc
+	     CC = $(TARGET_X)-gcc
 	     RM = rm -f
 	  MKDIR = mkdir -p
 	  RMDIR = rm -rf
@@ -48,14 +49,9 @@
 #
 #
 #
-SBINPRG =	killproc startproc checkproc
+SBINPRG =	killproc
 UBINPRG =
 
-ifeq ($(DISTRO),SuSE)
-   UBINPRG += usleep
-   UBINPRG += fsync
-endif
-
 all: $(SBINPRG) $(UBINPRG)
 
 libinit.o:	libinit.c libinit.h
@@ -64,107 +60,7 @@
 killproc:	killproc.c libinit.o
 	$(CC) $(CFLAGS) $(CLOOP) -o $@ $^ $(LIBS)
 
-startproc:	startproc.c libinit.o
-	$(CC) $(CFLAGS) $(CLOOP) -o $@ $^ $(LIBS)
-
-checkproc:	checkproc.c libinit.o
-	$(CC) $(CFLAGS) $(CLOOP) -o $@ $^ $(LIBS)
-
-usleep:		usleep.c
-	$(CC) $(CFLAGS) -o $@ $^
-
-fsync:		fsync.c
-	$(CC) $(CFLAGS) -o $@ $^
 
 clean:
-	$(RM) *.o *~ killproc startproc checkproc pidofproc start_daemon usleep fsync
+	$(RM) *.o *~ killproc
 
-install:	$(TODO)
-	if test -n "$(SBINPRG)" ; then	\
-	    $(MKDIR)	$(SBINDIR);	\
-	    $(MKDIR)	$(SDOCDIR);	\
-	fi
-	if test -n "$(UBINPRG)" ; then  \
-	    $(MKDIR)	$(UBINDIR);	\
-	    $(MKDIR)	$(UDOCDIR);	\
-	fi
-	for p in $(SBINPRG); do 		\
-	    $(INSTBIN) $$p	$(SBINDIR)/;	\
-	    $(INSTDOC) $$p.8	$(SDOCDIR)/;	\
-	done
-	if test -e $(SBINDIR)/startproc ; then	\
-	    $(LINK) startproc	$(SBINDIR)/start_daemon;	\
-	    $(LINK) startproc.8	$(SDOCDIR)/start_daemon.8;	\
-	fi
-	if test -e $(SBINDIR)/checkproc ; then	\
-	    $(LINK) checkproc	$(SBINDIR)/pidofproc;		\
-	    $(SO)checkproc.8 >	$(SDOCDIR)/pidofproc.8;		\
-	fi
-	for p in $(UBINPRG); do 		\
-	    $(INSTBIN) $$p	$(UBINDIR)/;	\
-	    $(INSTDOC) $$p.1	$(UDOCDIR)/;	\
-	done
-#
-# Make distribution
-#
-FILES	= README      \
-	  COPYING     \
-	  Makefile    \
-	  killproc.8  \
-	  killproc.c  \
-	  startproc.c \
-	  startproc.8 \
-	  checkproc.c \
-	  checkproc.8 \
-	  libinit.c   \
-	  libinit.h   \
-	  usleep.c    \
-	  usleep.1    \
-	  fsync.c    \
-	  fsync.1    \
-	  killproc-$(VERSION).lsm
-
-dest:
-	$(MKDIR) killproc-$(VERSION)
-	@echo -e 'Begin3\n\
-Title:		killproc and assorted tools for boot scripts\n\
-Version:	$(VERSION)\n\
-Entered-date:	$(DATE)\n\
-Description:	Some useful programs for a replacment of the shell functions\n\
-x 		daemom and killproc found in the Linux System V init suite.\n\
-x 		killproc(8) for signaling or terminating, checkproc(8) for\n\
-x 		checking and startproc(8) for starting processes.\n\
-x 		Each program has its own manual page.\n\
-Keywords:	killproc, startproc, checkproc, process control\n\
-Author:		Werner Fink <werner@suse.de>\n\
-Maintained-by:	Werner Fink <werner@suse.de>\n\
-Primary-site:	sunsite.unc.edu /pub/Linux/system/daemons/init\n\
-x		@UNKNOWN killproc-$(VERSION).tar.gz\n\
-Alternate-site:	ftp.suse.com /pub/projects/init\n\
-Platforms:	Linux with System VR2 or higher boot scheme\n\
-Copying-policy:	GPL\n\
-End' | sed 's@^ @@g;s@^x@@g' > killproc-$(VERSION).lsm
-	cp $(FILES) killproc-$(VERSION)
-	tar -c -zf  killproc-$(VERSION).tar.gz killproc-$(VERSION)/
-	$(RMDIR)    killproc-$(VERSION)
-	set -- `gzip -l killproc-$(VERSION).tar.gz | tail -1` ; \
-	sed "s:@UNKNOWN:$$1:" < killproc-$(VERSION).lsm > \
-	killproc-$(VERSION).lsm.tmp ; \
-	mv killproc-$(VERSION).lsm.tmp killproc-$(VERSION).lsm
-
-tar.gz: all
-	$(MKDIR) -p tmpr/sbin
-	$(MKDIR) -p tmpr/usr/share/man/man8
-	chown root:root tmpr/*
-	install -s -m 0755 -o root -g root killproc  tmpr/sbin/
-	install -s -m 0755 -o root -g root startproc tmpr/sbin/
-	install -s -m 0755 -o root -g root checkproc tmpr/sbin/
-	ln -sf startproc      tmpr/sbin/start_daemon
-	ln -sf checkproc      tmpr/sbin/pidofproc
-	gzip -c -9 killproc.8   > tmpr/usr/share/man/man8/killproc.8.gz
-	gzip -c -9 startproc.8  > tmpr/usr/share/man/man8/startproc.8.gz
-	gzip -c -9 checkproc.8  > tmpr/usr/share/man/man8/checkproc.8.gz
-	ln -sf startproc.8.gz tmpr/sbin/start_daemon.8.gz
-	ln -sf checkproc.8.gz tmpr/sbin/pidofproc.8.gz
-	cd tmpr/; tar cfsSpz ../killproc.tgz sbin/ usr/
-	rm -rf tmpr/
